analyzer/src/main/kotlin/curation/Sw360PackageCurationProvider.kt

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (C) 2020 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.analyzer.curation
+
+import java.io.File
+
+import org.eclipse.sw360.antenna.api.service.ServiceFactory
+import org.eclipse.sw360.antenna.sw360.client.adapter.SW360ConnectionFactory
+import org.eclipse.sw360.antenna.sw360.client.config.SW360ClientConfig
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.attachments.SW360AttachmentType
+import org.eclipse.sw360.antenna.sw360.client.rest.resource.releases.SW360SparseRelease
+
+import org.ossreviewtoolkit.analyzer.PackageCurationProvider
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.PackageCuration
+import org.ossreviewtoolkit.model.PackageCurationData
+import org.ossreviewtoolkit.model.RemoteArtifact
+import org.ossreviewtoolkit.model.Hash
+import org.ossreviewtoolkit.model.HashAlgorithm
+import org.ossreviewtoolkit.model.config.Sw360Configuration
+
+
+class Sw360PackageCurationProvider(sw360Configuration: Sw360Configuration?) : PackageCurationProvider {
+
+    private val sw360ConnectionConfig: SW360ClientConfig = SW360ClientConfig.createConfig(
+        sw360Configuration?.restUrl,
+        sw360Configuration?.authUrl,
+        sw360Configuration?.username,
+        sw360Configuration?.password,
+        sw360Configuration?.clientId,
+        sw360Configuration?.clientPassword,
+        ServiceFactory().createHttpClient(
+            sw360Configuration?.proxyHost.isNullOrEmpty() && sw360Configuration?.proxyPort!! > 0,
+            sw360Configuration?.proxyHost,
+            sw360Configuration?.proxyPort!!
+        ),
+        ServiceFactory.getObjectMapper()
+    )
```

Comment:
I expect that if the `sw360Configuration` is `null` this provider will not work. I would suggest to add an init block like this to the class to provide the user with meaningful feedback in this case:
```
init {
  requireNotNull(sw360Configuration) {
    "SW360 is not configured. To use curations from SW360 configure it in the ORT configuration file."
  }
}
```